source/OptimaJet.DWKit.StarterApplication/Repositories/UserRepository.cs

```diff
@@ -26,46 +26,47 @@ IDbContextResolver contextResolver
             this.CurrentUserContext = currentUserContext;
         }
 
-        protected IQueryable<User> DefaultGet()
-        {
-            return base.Get()
-                .Include(u => u.GroupMemberships)
-                    .ThenInclude(gm => gm.Group)
-                .Include(u => u.OrganizationMemberships)
-                    .ThenInclude(om => om.Organization);
-        }
         public override IQueryable<User> Get()
         {
-            if (OrganizationContext.SpecifiedOrganizationDoesNotExist) return Enumerable.Empty<User>().AsQueryable();
+            if (OrganizationContext.SpecifiedOrganizationDoesNotExist) {
+                return Enumerable.Empty<User>().AsQueryable();
+            }
+
+            var currentUser = GetByAuth0Id(CurrentUserContext.Auth0Id).Result;
+
+            var query = base.Get();
+
             if (!OrganizationContext.HasOrganization)
             {
-                // No organization specified, so include all users in the all the organizations that the current user is a member
-                var currentUser = GetByAuth0Id(CurrentUserContext.Auth0Id).Result;
-                return DefaultGet()
-                           .Where(u => u.OrganizationMemberships.Select(o => o.OrganizationId).Intersect(currentUser.OrganizationIds).Any());
-            }
-            // Get users in the current organization
-            return DefaultGet()
-                       .Where(u => u.OrganizationMemberships.Any(o => o.OrganizationId == OrganizationContext.OrganizationId));
-        }
+                // No organization specified, so include all users in the all the 
+                // organizations that the current user is a member
+                var orgIds = currentUser.OrganizationIds ?? Enumerable.Empty<int>();
 
-        public override async Task<User> GetAsync(int id)
```

Comment:
I got rid of GetAsync(id), because Its implementation relies on the Get(), so I moved the inclusion of the current-user into there.